pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -0,0 +1,573 @@
+		}
+		return newPCCloud(cfg)
+	})
+}
+
+// Retrieve the Photon VM ID from the Photon Controller endpoint based on the node name
+func getVMIDbyNodename(project string, nodeName string) (string, error) {
+	vmList, err := photonClient.Projects.GetVMs(project, nil)
+	if err != nil {
+		return "", fmt.Errorf("Failed to GetVMs from project %s with nodeName %s, error: [%v]", project, nodeName, err)
+	}
+
+	for _, vm := range vmList.Items {
+		if vm.Name == nodeName {
+			if vm.State == "STARTED" {
+				return vm.ID, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No matching started VM is found with name %s", nodeName)
+}
+
+// Retrieve the Photon VM ID from the Photon Controller endpoint based on the IP address
+func getVMIDbyIP(project string, IPAddress string) (string, error) {
+	vmList, err := photonClient.Projects.GetVMs(project, nil)
+	if err != nil {
+		return "", fmt.Errorf("Failed to GetVMs for project %s, error [%v]", project, err)
+	}
+
+	for _, vm := range vmList.Items {
+		task, err := photonClient.VMs.GetNetworks(vm.ID)
+		if err != nil {
+			glog.Warningf("Photon Cloud Provider: GetNetworks failed for vm.ID %s, error [%v]", vm.ID, err)
+		} else {
+			task, err = photonClient.Tasks.Wait(task.ID)
+			if err != nil {
+				glog.Warning("Photon Cloud Provider: Wait task for GetNetworks failed for vm.ID %s, error [%v]", vm.ID, err)
+			} else {
+				networkConnections := task.ResourceProperties.(map[string]interface{})
+				networks := networkConnections["networkConnections"].([]interface{})
+				for _, nt := range networks {
+					network := nt.(map[string]interface{})
+					if val, ok := network["ipAddress"]; ok && val != nil {
+						ipAddr := val.(string)
+						if ipAddr == IPAddress {
+							return vm.ID, nil
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No matching VM is found with IP %s", IPAddress)
+}
+
+// Retrieve the the Photon project ID from the Photon Controller endpoint based on the project name
+func getProjIDbyName(tenantName, projName string) (string, error) {
+	tenants, err := photonClient.Tenants.GetAll()
+	if err != nil {
+		return "", fmt.Errorf("GetAll tenants failed with error [%v]", err)
+	}
+
+	for _, tenant := range tenants.Items {
+		if tenant.Name == tenantName {
+			projects, err := photonClient.Tenants.GetProjects(tenant.ID, nil)
+			if err != nil {
+				return "", fmt.Errorf("Failed to GetProjects for tenant %s, error [%v]", tenantName, err)
+			}
+
+			for _, project := range projects.Items {
+				if project.Name == projName {
+					return project.ID, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No matching tenant/project name is found with %s/%s", tenantName, projName)
+}
+
+func newPCCloud(cfg PCConfig) (*PCCloud, error) {
+	if len(cfg.Global.CloudTarget) == 0 {
+		return nil, fmt.Errorf("Photon Controller endpoint was not specified.")
+	}
+
+	//TODO: add handling of certification enabled situation
+	options := &photon.ClientOptions{
```

Comment:
Does this mean it is insecure?  Please comment on what this means (and make sure it's in the docs too)